### ISSUE TITLE
Native binaries in Extensions missing `+x`

### ISSUE THREAD

**Issue by thrimbor** [NationalSecurityAgency/ghidra#3448]
**Describe the bug**
When installing an extension that contains native binaries, the native binaries sometimes are missing the execute permission. This makes analysis steps that try to run them fail.
I've had this issue before (https://github.com/mborgerson/ghidra-xbe/issues/27), but it seemed to not appear anymore, making me think it was a one off fluke, but I can now reproduce it again.

**To Reproduce**
1. Download and install the extension linked below
2. Check the permissions for `ghidra-xbe/os/linux64/XbSymbolDatabaseTool`. `+x` will be missing.
3. Optionally, download the xbe in https://github.com/mborgerson/ghidra-xbe/tree/master/tests/xbefiles and try to analyze it in Ghidra. The symbol analyzer step should fail.

**Expected behavior**
`ghidra-xbe/os/linux64/XbSymbolDatabaseTool` should have the execute permission, and running the symbol detection step on an xbe file should succeed.

**Screenshots**
![Screenshot_20210924_232426](https://user-images.githubusercontent.com/1339483/134743262-eb14ebee-ca0c-4e3e-9d92-91b2be308c5f.png)

**Attachments**
[ghidra_10.0.3_PUBLIC_20210924_ghidra-xbe.zip](https://github.com/NationalSecurityAgency/ghidra/files/7228552/ghidra_10.0.3_PUBLIC_20210924_ghidra-xbe.zip)
[log.txt](https://github.com/NationalSecurityAgency/ghidra/files/7228564/log.txt) (log file containing extensions installation and usage, the other analysis steps failing is unimportant)

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Java Version: OpenJDK 16.0.2
 - Ghidra Version: 10.0.3
 - Ghidra Origin: Arch Linux repositories

**Additional context**

**Comment by agarmash** [comment]
Faced the same issue with Ghidra 10.0.3 on macOS 11.4.  

After doing some research, I figured out that there's nothing wrong with Ghidra - it preserves file system permissions while unpacking the extension.  

To verify that, I did the following:
 - Unpacked the `ghidra_10.0.3_PUBLIC_20210924_ghidra-xbe.zip` file with `unzip`;
 - Gave `XbSymbolDatabaseTool` permission for executing with `chmod +x`;
 - Repacked the extension with `zip -r`.  

After installing the modified extension, the aforementioned binary had all the required permissions.

**Comment by thrimbor** [comment]
@agarmash you seem to be correct, the change you submitted to ghidra-xbe fixes the issue. I suspected Ghidra might preserve the permissions, but I must've made a mistake during testing, because the zip file didn't contain the execute permission flag when I set it manually on the filesystem and rebuilt.

I'm also not sure why this issue didn't occur every time, but since it's fixed now and doesn't appear to be a Ghidra bug, I think I can close this.
